### PR TITLE
[2/2] Retrieve video submissions in timeline algorithms

### PR DESCRIPTION
**Pull request by louietyj**
Treat an attempted video as "submitted", i.e. do not modify its personal times anymore. This is to prevent subsequent recomputations from shifting the opening date later, which means that the student suddenly gets denied access to it.

**Test Plan**

Make a video attempt (Lecture 1B), create a personal time for the video that is way off, recompute and make sure the personal time for that video is untouched.

image

image

### CODE REVIEW

File: app/controllers/concerns/course/lesson_plan/personalization_concern.rb [hunk 1]
```diff
@@ -158,12 +170,12 @@ def retrieve_submissions(course_user)
...
   # order to have any hope of completing the course within the limits. The algorithm needs to take care of this.
   def compute_learning_rate_effective_limits(
-    course_user, items, submitted_ids, min_overall_learning_rate, max_overall_learning_rate
+    course_user, items, submissions, min_overall_learning_rate, max_overall_learning_rate
   )
     course_start = items.first.start_at
     course_end = items.last.start_at
     last_submitted_item =
-      items.reverse_each.lazy.select { |item| item.affects_personal_times? && item.id.in?(submitted_ids) }.first
+      items.reverse_each.lazy.select { |item| item.affects_personal_times? && item.id.in?(submissions.keys) }.first
```
Comment (trangmei) [inline app/controllers/concerns/course/lesson_plan/personalization_concern.rb]: `submissions` can be an empty array

Comment (trangmei) [reply app/controllers/concerns/course/lesson_plan/personalization_concern.rb]: Just pulled `personal-times-playground` directly, can no longer reproduce

Comment (louietyj) [reply app/controllers/concerns/course/lesson_plan/personalization_concern.rb]: From eyeballing the `compute_learning_rate_effective_limits`, even if there are no submissions it should be an empty hash (hash merge with hash merge with hash), so I'm not sure why you're getting an array. Might have just been a hot-reloading fluke 😂

Lmk if you can repro again and I'll look into it.

**Comment by louietyj** [comment]
Oh yeah good point, will do that

**Comment by fonglh** [comment]
`submissions` here are actually `submitted_lesson_plan_items` right? Can we change the variable names to reflect that?

It's not going to be immediately obvious a few months later or when someone else comes to take a look that they're not actually `Course::Assessment::Submission` or `Course::Video::Submission`

**Comment by louietyj** [comment]
That would be similarly misleading because there is no Course::LessonPlan::Item there either. It's actually a hash of `lesson_plan_item_id -> submission_time or nil` as documented on the method itself, which will show up if you hover over the method name.

I can do `retrieve_lesson_plan_items_submission_time_hash` with the trade-off that the method name takes up nearly half the line width limit on rubocop.

**Comment by fonglh** [comment]
It's less misleading as the model type is correct at least. `submissions` is used pretty much everywhere else to refer to the 2 submission models. The hovering tip is specific to your code editor setup, so we shouldn't assume that everyone sees the same tooltip.

Drop the `retrieve`, you can see existing names to see how they're named. From the root folder:

```
grep -ri 'def .*_hash' app
```

**Pull request merged**